java-surefire/src/main/java/org/sonar/plugins/surefire/SurefireJavaParser.java

```diff
@@ -164,8 +178,19 @@ protected void saveResults(InputFile testFile, UnitTestClassReport report) {
     }
   }
 
-  protected InputFile getUnitTestResource(String classKey) {
-    return javaResourceLocator.findResourceByClassName(classKey);
+  @CheckForNull
+  private InputFile getUnitTestResource(String className, UnitTestClassReport unitTestClassReport) {
+    InputFile resource = javaResourceLocator.findResourceByClassName(className);
+    if (resource == null) {
+      // fall back on testSuite class name (repeated and parameterized tests from JUnit 5.0 are using test name as classname)
+      // Should be fixed with JUnit 5.1, see: https://github.com/junit-team/junit5/issues/1182
+      return unitTestClassReport.getResults().stream()
```

Comment:
two things I struggle to see through or understand are : 
- How will this behave on junit 4 parameterized test ? 
- How will this behave when fix is provided in 5.1 ?

Reply:
These aspects are indeed missing.
* I'll add a test for junit 4 parameterized tests.
* I'll add a "fake" test case in the report simulating fix of 5.1 and a correct class name.